### DISCUSSION TITLE
Restructure the project a little.

### DISCUSSION THREAD

**Discussion by ianwijma** [Heroic-Games-Launcher/HeroicGamesLauncher#210] [Ideas]
Initially I was a bit confused where was what and wasn't 100% sure why. But after looking at the directory structure, I found out that some files where in unexpected places, like:

- the `public/` folder contains `start-react.js` that actually starts electron after react.
- Source code is outside the `src/` folder, take the `electron/` folder for example.
- The `public/` folder contains static and generated code. Which really confused me in the beginning when my edited code was overwritten.
- ContextProvider in the `state/` folder threw me off.
- Static assets in the `src/` folder

These are a few of the items that confused me initially when working on the project. So I'd decided to sit down and design a new directory structure as a suggestion for when we are moving to the new UI design. I have written a mockup structure over at my repository over here: [directoryDesign](https://github.com/ianwijma/directoryDesign)

I'm not 100% sure if everything is possible, because I never works on a Electron + React + TS combo. But I'd think the project benefits if we organise the folder structure a bit where everything has its own folder to make it easier to navigate and understand.

**Comment by flavioislima** [comment]
I think that would be good, I agree some folders can be confusing. Also, some components needs to be refactored into smaller ones.

So, the first thing we need to have in mind is that Electron has some peculiarities, and since we are using it with typescript, some arrangements are needed, so that's why we have the electron folder and a tsconfig for it.
The public folder needs to exist because all React projects and Electron needs it since the build content generated for the binary, stays in this folder.

Maybe the Electron folder could be moved into src, or we could have folders for backend and frontend.
The context provider is the state management we are using for Heroic, I didn't want to go with Redux since I dont like its complexity for small projects. Since the state management now is small, I think we are fine using Context. Also, we can have another context for other parts of the code. I'm thinking in creating one for the Settings to simplify that file that is becoming huge and complex.

The Assets needs to be inside a folder that React can see because REact cant access anything outside the src dir, this is a global React rule and, even though can be broken with some workarounds, is not recommended.
So assets could go inside the frontend folder that will contain the react code.

Other than that I think the idea is good and we can start doing it little by little to avoid breaking stuff. :)

**Reply by adityaruplaha** [reply]
I agree, this needs to happen.
